src/qt/walletmodel.cpp

```diff
@@ -591,6 +594,24 @@ bool WalletModel::isSpent(const COutPoint& outpoint) const
     return wallet->IsSpent(outpoint.hash, outpoint.n);
 }
 
+bool WalletModel::isUsed(const std::string& address) const
+{
+    LOCK2(cs_main, wallet->cs_wallet);
+    for (const std::pair<CTxDestination, CAddressBookData>& item : wallet->mapAddressBook)
```

Comment:
Instead of looping, would a std::map::find be possible (`std::map<CTxDestination, CAddressBookData>::iterator mi = wallet->mapAddressBook.find(dest);`?